cmd/panda_test_executor/panda_executor_test.go

```diff
@@ -0,0 +1,456 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	controller "github.com/panda-re/panda_studio/internal/panda_controller"
+)
+
+// Each enum represents the state panda was in that caused the exception
+// Enum should match that in /panda_agent/agent.py
+const (
+	RUNNING       = 0
+	NOT_RUNNING   = 1
+	RECORDING     = 2
+	NOT_RECORDING = 3
+	REPLAYING     = 4
+	NOT_REPLAYING = 5
+)
+
+var error_to_string [6]string = [6]string{"RUNNING", "NOT_RUNNING", "RECORDING", "NOT_RECORDING", "REPLAYING", "NOT_REPLAYING"}
+
+// Extracts the error number from an err from agent
+func getError(err error) int {
+	// Find the numbers in the error message
+	re := regexp.MustCompile("[0-9]+")
+	// Return the first one as an integer
+	nums := re.FindAllString(err.Error(), -1)
+	num, _ := strconv.Atoi(nums[0])
+	return num
+}
+
+var num_passed int = 0
+var num_tests int = 0
+
+// Runs a test for the agent, recording and replay
+// Prints the number of tests and success rate
+func TestMain(t *testing.T) {
+	t.Cleanup(func() {
+		fmt.Printf("Number of tests: %d\nNumber passed: %d\nSuccess rate: %d%%\n", num_tests, num_passed, 100*num_passed/num_tests)
+	})
+	t.Run("Agent", TestAgent)
+	if t.Failed() {
+		t.Fatal("Agent unsuccessful")
+	}
+	t.Run("Recording", TestRecord)
+	if t.Failed() {
+		t.Fatal("Recording unsuccessful")
+	}
+	t.Run("Replay", TestReplay)
```

Comment:
Does it matter if you test if the replay failed here? Or is that accounted for in the cleanup?

Reply:
The other tests are checked for failure because if the previous test failed, then the next test may fail or may not have everything it needs (like a recording to replay). I'm 20/80 as to whether we should keep those checks or remove them and let the next tests possibly fail and run the entire test suite regardless.

Reply:
We can make that change later if we want to. I'll merge it in the meantime